### ISSUE TITLE
Support for extended classes

### ISSUE THREAD

**Issue by jrust** [christianalfoni/immutable-store#8]
I was trying to use [class-extender](https://github.com/seeden/class-extender) to create child classes and then store instances of a child class in an immutable store.  But it turns out immutable-store only ends up keeping properties from the parent class.

Is it possible to support more than object literals in the immutable store?  If not, or it would be too complex, it may be good to mention in the README that inherited classes aren't supported.

**Comment by christianalfoni** [comment]
Hi @jrust ,

Yeah, actually: "Put plain objects, arrays and primitives in the store." is mentioned in the README, but probably not good enough. Do you see a fitting place to highlight it?

It would be possible to support objects with their own prototype, but that kinda opens up a path I would like to avoid. For example handling files etc. would not work with this approach.

So yeah, I would prefer keeping it to base JavaScript types.

I am totally open to discuss this more if you have a use case where this causes challenges :-)

**Comment by jrust** [comment]
Thanks for the update @christianalfoni! I did some more reading and I think you're right -- classes, with their goal of providing an interface to encapsulated state, inheritance, and some other OOP constructs is largely orthogonal to immutable _data structures_, which seems to spring from the functional programming world.  Also I played around with immutable.js and it also seems to only support a limited number of data structures, and newed object are not one of them (as an aside, after experimenting with that library, I like immutable-store all the more.  It's really nice keep the native JS API for reading from the stores!).

I think the supported structures could be re-iterated under API.  I'll put together a PR for that.
